### DISCUSSION TITLE
StiffenedGasTwoPhaseFluidProperties problem

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#26431] [Q&A General] [answered]
Hi all,

I would like to use StiffenedGasTwoPhaseFluidProperties to calculate the material properties of two-phase flow.
I defined fp like this:
 
in .h file:
```
  const TwoPhaseFluidProperties & _fp;
  const SinglePhaseFluidProperties & _fp_liquid;
  const SinglePhaseFluidProperties & _fp_vapor;
```

in .C file:
```
  _fp(getUserObject<TwoPhaseFluidProperties>("fp")),
  _fp_liquid(getUserObject<SinglePhaseFluidProperties>(_fp.getLiquidName())),
  _fp_vapor(getUserObject<SinglePhaseFluidProperties>(_fp.getVaporName()))

and 

  _rho_liq[_qp] = _fp_liquid.rho_from_p_T(1.5e6,400.0);
  _rho_vap[_qp] = _fp_vapor.rho_from_p_T(1.8e6,300.0);

```

in the input file:
```
[FluidProperties]
  [./two_phase]
    type = StiffenedGasTwoPhaseFluidProperties
  [../]
```

And then some error occurred:
![image](https://github.com/idaholab/moose/assets/118275988/4981f0dd-0f69-4ffc-902b-0354787c96d6)

Is there something wrong?

Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

getUserObject takes the parameter name not the object name.
getUserObjectByName, if it exists, would take the object name like you have now "_fp.getLiquidName()"

Guillaume

**Reply by SomnusYu** [reply]
Thanks a lot, it works!
